server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -352,27 +350,11 @@ public void handleException(TransportException exp) {
                         onCompletionListener.onResponse(response);
                     }, e -> handleException(primaryShardReference, e));
 
-                    final ActionListener<Response> globalCheckpointSyncingListener = ActionListener.wrap(response -> {
-                        if (syncGlobalCheckpointAfterOperation) {
-                            final IndexShard shard = primaryShardReference.indexShard;
-                            try {
-                                shard.maybeSyncGlobalCheckpoint("post-operation");
-                            } catch (final Exception e) {
-                                // only log non-closed exceptions
-                                if (ExceptionsHelper.unwrap(
-                                    e, AlreadyClosedException.class, IndexShardClosedException.class) == null) {
-                                    // intentionally swallow, a missed global checkpoint sync should not fail this operation
-                                    logger.info(
-                                        new ParameterizedMessage(
-                                            "{} failed to execute post-operation global checkpoint sync", shard.shardId()), e);
-                                }
-                            }
-                        }
-                        referenceClosingListener.onResponse(response);
-                    }, referenceClosingListener::onFailure);
+                    final ActionListener<Response> responseListener = getResponseActionListener(referenceClosingListener,
```

Comment:
I think I would prefer this to be:

```suggestion
                    final ActionListener<Response> responseListener = ActionListener.map(this::maybeSyncGlobalCheckpoint, referenceClosingListener);
```

and `getResponseActionListener should then not do the wrapping.

Alternatively rename `getResponseActionListener` to something like `wrapSyncGlobalCheckpoint`.